ast-nodes/sum.dart

```diff
@@ -1,9 +1,29 @@
+import '../iterator-utils.dart';
+import '../syntax-error.dart';
+import 'add-operator.dart';
 import 'comparison.dart';
 import '../lexer.dart';
+import 'product.dart';
+import 'sub-operator.dart';
 
 /// An abstract summing operator.
 abstract class Sum implements Comparison {
   factory Sum.parse(Iterable<Token> tokens) {
-
+    final iter = tokens.iterator;
+    iter.moveNext();
+    final productBuffer = consumeUntil(iter, RegExp('^[+-]\$'));
```

Comment:
You should probably be aware that `-` is also a token for the `Primary` (the unary minus operator), which you should be able to recognize and leave in place